frontend/src/pages/settings/profile/Index.tsx

```diff
@@ -0,0 +1,105 @@
+import { Container, Stack, Avatar, Typography, Button, FormControl } from "@mui/material";
+import { TextFieldElement, FormContainer } from "react-hook-form-mui";
+import { useSelector } from "react-redux";
+import { selectUser } from "../../../store/userSlice";
+import { useEffect, useMemo, useState } from "react";
+import { useDebounce } from "react-use";
+import { useUpdateUserNicknameMutation } from "../../../hooks/api/user";
+import { useCheckNameConflictQuery } from "../../../hooks/api/check";
+
+const avatarSize = 117;
+
+function ProfileIndex() {
+	const userStore = useSelector(selectUser);
+	const [nickname, setNickname] = useState(userStore.data?.nickname || "");
+	const [debouncedNickname, setDebouncedNickname] = useState("");
+	const { data: conflictResult } = useCheckNameConflictQuery(debouncedNickname);
+	const { mutateAsync: updateUserNickname } = useUpdateUserNicknameMutation();
+	const errorMessage = useMemo(() => {
+		if (debouncedNickname != userStore.data?.nickname && conflictResult?.conflict) {
+			return "Already Exists";
+		}
+		return null;
+	}, [conflictResult?.conflict, debouncedNickname, userStore.data?.nickname]);
+
+	const isSubmit = () => {
+		return (
+			Boolean(errorMessage) || nickname === userStore.data?.nickname || nickname.length === 0
+		);
+	};
```

Comment:
The variable name and content seem to be reversed.

- Submission possible -> `false`
- Submission not possible -> `true`

Reply:
i'm very sorry. 

rename `isSubmit` to `isSubmitDisabled`